File: app/src/main/java/com/meteorhead/leave/mainactivity/di/ActivityModule.java
package com.meteorhead.leave.mainactivity.di;

import com.meteorhead.leave.mainactivity.ActivityViewModel;

import dagger.Module;
import dagger.Provides;

/**
 * Created by wierzchanowskig on 26.11.2016.
 */
@Module
public class ActivityModule {

    @ActivityScope
    @Provides
    ActivityViewModel provideActivityViewModel() {
        return new ActivityViewModel();
    }
}

File: app/src/main/java/com/meteorhead/leave/mainactivity/di/ActivityModule.kt
package com.meteorhead.leave.mainactivity.di

import com.meteorhead.leave.mainactivity.ActivityViewModel

import dagger.Module
import dagger.Provides

/**
 * Created by wierzchanowskig on 26.11.2016.
 */
@Module
class ActivityModule {

    @ActivityScope
    @Provides
    internal fun provideActivityViewModel(): ActivityViewModel {
        return ActivityViewModel()
    }
}
